src/plugins/vis_default_editor/public/components/controls/filters.tsx

```diff
@@ -40,10 +40,16 @@ function FiltersParamEditor({ agg, value = [], setValue }: AggParamEditorProps<F
     value.map(filter => ({ ...filter, id: generateId() }))
   );
 
+  const firstRender = useRef(true);
+
   useEffect(() => {
-    // set parsed values into model after initialization
-    setValue(filters.map(filter => omit({ ...filter, input: filter.input }, 'id')));
-  }, []);
+    if (firstRender.current) {
```

Comment:
So this `firstRender` workaround is basically an alternative to disabling linting like in https://github.com/elastic/kibana/pull/62475/files?diff=unified&w=1#diff-68df3519b81dd4040af6faf4820bd3b7R101 , right?

Why are you using both ways in different places? Subjectively I prefer the comment disabling linting for now, this seems dirtier to me.

Reply:
You are right, it looks not really in place here and breaks up the React Hooks essentials,
but I wouldn't also use the eslint disabling (the one you mentioned was just accidentally missed, my fault).
I'm not sure how soon we'll be able to refactor this code due to we have a list of blockers and to-dos first to make the default editor more pure and meet meet all of React guidelines (here I mean agg configs restructure, agg types, etc).
In that case I would consider using more explicit helper - `useMount` from `react-use` , so this will look like :

```ts
import { useMount } from 'react-use';

...

  useMount(() => {
    // set parsed values into model after initialization
    setValue(filters.map(filter => omit({ ...filter, input: filter.input }, 'id')));
  });
```

What do you think ?

Reply:
+1 for this, as we are using `react-use` in a bunch of places it sounds like a solution that achieves the same hiding the workaround somewhere else.

Reply:
Updated!